src/domain/entities/schemas/lenient-page.schema.factory.ts

```diff
@@ -0,0 +1,21 @@
+import { TypeOf, z } from 'zod';
+import { Page } from '@/domain/entities/page.entity';
+
+/**
+ * Builds a lenient page schema that filters out invalid items from the results array.
+ */
+export function buildLenientPageSchema<T extends z.ZodTypeAny>(
+  itemSchema: T,
+): z.ZodType<Page<z.infer<T>>> {
+  return z.object({
+    count: z.number().nullable(),
+    next: z.string().nullable(),
+    previous: z.string().nullable(),
+    results: z.array(z.any()).transform((items) =>
+      items.flatMap((item) => {
+        const result = itemSchema.safeParse(item);
+        return result.success ? [result.data as TypeOf<T>] : [];
```

Comment:
Do you think we should log details of those items that failed validation? We also don't need to cast `result.data`.

Reply:
Those are being logged here (on the caller): https://github.com/safe-global/safe-client-gateway/blob/17054280c68c28802ff5ea29131c418a48251b74/src/domain/chains/chains.repository.ts#L41

I thought it was better to let the caller decide about logging than doing it into the schema builder function, wdyt?

Reply:
Makes sense to me!